test/javascript/components/track/IterationSummary.test.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import { render, screen } from '@testing-library/react'
+import '@testing-library/jest-dom/extend-expect'
+import { IterationSummary } from '../../../../app/javascript/components/track/IterationSummary'
+
+test('shows details', async () => {
+  render(
+    <IterationSummary
+      iteration={{
+        idx: 2,
+        submissionMethod: 'cli',
+        createdAt: Date.now() - 1,
+        testsStatus: 'queued',
+        numEssentialAutomatedComments: 2,
+      }}
+    />
+  )
+
+  expect(screen.getByText('Iteration 2')).toBeInTheDocument()
+  expect(screen.getByTitle('Submitted via CLI')).toBeInTheDocument()
+  expect(screen.getByTestId('details')).toHaveTextContent(
+    'Submitted via CLI, a few seconds ago'
+  )
+  expect(
+    screen.getByRole('status', { name: 'Processing status' })
+  ).toHaveTextContent('Passed')
+  expect(
+    screen.getByRole('status', { name: 'Analysis status' })
+  ).toHaveTextContent('2')
```

Comment:
@iHiD I don't exactly what the status is of the latest JS tests. In other words: does this correspond to how we should now we writing the tests to prevent any testing issues?

Reply:
Yeah - this looks good to me. The only thing might be adding an `await awaitPopper()` above the first expect if this component uses popper.

But because there's no interactivity, we don't need any waitFor or await blocks after that initial render.